src/Lecture4.hs

```diff
@@ -184,7 +218,19 @@ instance for the 'Stats' type itself.
 -}
 
 instance Semigroup Stats where
-
+  (<>) :: Stats -> Stats -> Stats
+  (<>) (Stats a b c d e f g h i) (Stats a2 b2 c2 d2 e2 f2 g2 h2 i2) =
+    Stats (a <> a2) (b <> b2) (c <> c2) (d <> d2) (c' e e2) (c' f f2) (c' g g2) (c' h h2) (i <> i2)
+-- hmm
+-- just want to pairwise zip the args
+-- could gmappend from Data.Semigroup.Generic work?
+--   (<>) = gmappend
```

Comment:
Yes, it could work 🙂 
But you also need to derive `Generic` first (which is usually not a problem). However, you won't be able to implement the desired strict behaviour for `Maybe` with `gmappend`